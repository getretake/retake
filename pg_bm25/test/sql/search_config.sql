-- Basic seach query
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'category:electronics';
-- With trailing delimiter
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'category:electronics:::';
-- With limit
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'category:electronics' LIMIT 2;
-- With limit and offset
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'category:electronics:::offset=1' LIMIT 2;
-- With fuzzy field
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'category:electornics:::fuzzy_fields=category';
-- Without fuzzy field
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'category:electornics';
-- With fuzzy field and transpose_cost_one=false and distance=1
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'keybaord:::fuzzy_fields=description&transpose_cost_one=false&distance=1';
-- With fuzzy field and transpose_cost_one=true and distance=1
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'keybaord:::fuzzy_fields=description&transpose_cost_one=true&distance=1';
-- With fuzzy and regex field
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'com:::regex_fields=description&fuzzy_fields=description';
-- With regex field 
SELECT id, description, rating, category FROM search_config WHERE search_config @@@ 'com:::regex_fields=description';
-- Default highlighting without max_num_chars
SELECT description, rating, category, paradedb.highlight_bm25(ctid, 'idxsearchconfig', 'description') FROM search_config WHERE search_config @@@ 'description:keyboard OR category:electronics' ORDER BY paradedb.rank_bm25(ctid) DESC LIMIT 5;
-- max_num_chars is set to 14 
SELECT description, rating, category, paradedb.highlight_bm25(ctid, 'idxsearchconfig', 'description') FROM search_config WHERE search_config @@@ 'description:keyboard OR category:electronics:::max_num_chars=14' ORDER BY paradedb.rank_bm25(ctid) DESC LIMIT 5;
