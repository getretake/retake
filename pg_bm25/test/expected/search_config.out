-- With fuzzy field
SELECT id, description, rating, category FROM search_config.search('category:electornics', fuzzy_fields => 'category');
 id |         description         | rating |  category   
----+-----------------------------+--------+-------------
  1 | Ergonomic metal keyboard    |      4 | Electronics
  2 | Plastic Keyboard            |      4 | Electronics
 12 | Innovative wireless earbuds |      5 | Electronics
 22 | Fast charging power bank    |      4 | Electronics
 32 | Bluetooth-enabled speaker   |      3 | Electronics
(5 rows)

-- Without fuzzy field
SELECT id, description, rating, category FROM search_config.search('category:electornics');
 id | description | rating | category 
----+-------------+--------+----------
(0 rows)

-- With fuzzy field and transpose_cost_one=false and distance=1
SELECT id, description, rating, category FROM search_config.search('description:keybaord', fuzzy_fields => 'description', transpose_cost_one => false, distance => 1);
 id | description | rating | category 
----+-------------+--------+----------
(0 rows)

-- With fuzzy field and transpose_cost_one=true and distance=1
SELECT id, description, rating, category FROM search_config.search('description:keybaord', fuzzy_fields => 'description', transpose_cost_one => true, distance => 1);
 id |       description        | rating |  category   
----+--------------------------+--------+-------------
  1 | Ergonomic metal keyboard |      4 | Electronics
  2 | Plastic Keyboard         |      4 | Electronics
(2 rows)

-- With regex field 
SELECT id, description, rating, category FROM search_config.search('com', regex_fields => 'description');
 id |      description       | rating |  category   
----+------------------------+--------+-------------
  6 | Compact digital camera |      5 | Photography
 23 | Comfortable slippers   |      3 | Footwear
(2 rows)

-- Default highlighting without max_num_chars
SELECT description, rating, category, highlight_bm25 FROM search_config.search('description:keyboard OR category:electronics') as s LEFT JOIN search_config.highlight('description:keyboard OR category:electronics', highlight_field => 'description') as h ON s.id = H.id LEFT JOIN search_config.rank('description:keyboard OR category:electronics') as r ON s.id = r.id ORDER BY rank_bm25 DESC LIMIT 5;
         description         | rating |  category   |         highlight_bm25          
-----------------------------+--------+-------------+---------------------------------
 Plastic Keyboard            |      4 | Electronics | Plastic <b>Keyboard</b>
 Ergonomic metal keyboard    |      4 | Electronics | Ergonomic metal <b>keyboard</b>
 Innovative wireless earbuds |      5 | Electronics | 
 Fast charging power bank    |      4 | Electronics | 
 Bluetooth-enabled speaker   |      3 | Electronics | 
(5 rows)

-- max_num_chars is set to 14 
SELECT description, rating, category, highlight_bm25 FROM search_config.search('description:keyboard OR category:electronics', max_num_chars => 14) as s LEFT JOIN search_config.highlight('description:keyboard OR category:electronics', highlight_field => 'description', max_num_chars => 14) as h ON s.id = H.id LEFT JOIN search_config.rank('description:keyboard OR category:electronics', max_num_chars => 14) as r ON s.id = r.id ORDER BY rank_bm25 DESC LIMIT 5;
         description         | rating |  category   |    highlight_bm25     
-----------------------------+--------+-------------+-----------------------
 Plastic Keyboard            |      4 | Electronics | <b>Keyboard</b>
 Ergonomic metal keyboard    |      4 | Electronics | metal <b>keyboard</b>
 Innovative wireless earbuds |      5 | Electronics | 
 Fast charging power bank    |      4 | Electronics | 
 Bluetooth-enabled speaker   |      3 | Electronics | 
(5 rows)

